Add tls section to S3 concepts guide (#550)
stackabletech/documentation: modules/concepts/pages/s3.adoc

        pod: {}
----

Reference it from the connection object:

[source,yaml]
----
...
host: test-minio
port: 9000
credentials:
  secretClass: s3-credentials-class
...
----

== TLS

TLS certificates are specified in a similar way. You will need a `Secret` containing the certificate, a `SecretClass` and then a reference to this `SecretClass` where you want to specify the certificate.

The Secret must contain the TLS certificate and the CA certificate. The `tls.crt` and `tls.key` are the certificate and the private key of the server. The `ca.crt` is the certificate of the Certificate Authority that signed the server's certificate. The values are expected to be base64 encoded PEM files.

[source,yaml]
----
apiVersion: v1
kind: Secret
metadata:
  name: s3-certificates
  labels:
    secrets.stackable.tech/class: s3-certificate-class
data:
  ca.crt: ...
  tls.crt: ...

>   tls.key: ...

Reviewer: You actually only need the ca.crt. WDYT of including the SecretClass as well?

Author: Updated https://github.com/stackabletech/documentation/pull/550/commits/ddb926a3c8f8f24e5b8cd5e84e2434a27b4bd7f8

        pod: {}
----

Reference it from the connection object:

[source,yaml]
----
...
host: test-minio
port: 9000
credentials:
  secretClass: s3-credentials-class
...
----

== TLS

> TLS certificates are specified in a similar way. You will need a `SecretClass` that is referenced by the connection object. When using the `k8sSearch` backend, a `Secret` object containing the certificate/key pair is also needed.

Reviewer: ```suggestion
TLS certificates are specified in a similar way. You will need a `SecretClass` that is referenced by the connection object. When using the `k8sSearch` backend, a `Secret` object containing the ca certificate is also needed.
```

> The `Secret` must contain the TLS certificate and the CA certificate. The `tls.crt` and `tls.key` are the certificate and the private key of the server. The `ca.crt` is the certificate of the Certificate Authority that signed the server's certificate. This is usually optional. The values are expected to be base64 encoded PEM files.

Reviewer: ```suggestion
The Secret must contain the CA certificate in the form of a `ca.crt` key, which contains the Certificate Authority that signed the server's certificate. The values are expected to be base64 encoded PEM files.
```

In this case, the `SecretClass` would look like this:

[source,yaml]
----
apiVersion: secrets.stackable.tech/v1alpha1
kind: SecretClass
metadata:
  name: s3-certificate-class
spec:
  backend:
    k8sSearch:
      searchNamespace:
        pod: {}
----

And the `Secret` object defining the certificate/key pair:

[source,yaml]
----
apiVersion: v1
kind: Secret
metadata:
  name: s3-certificates
  labels:
    secrets.stackable.tech/class: s3-certificate-class
data:
  ca.crt: ...

>   tls.crt: ...
>   tls.key: ...

Reviewer: I think I would only include `ca.crt`, as often times I only have my company root ca, but no certificate at all. I a s a user would be "why do I need a certificate to access my S3`"
```suggestion
```
